File: libs/frontend/shared/schema/src/lib/knowledge-graph.ts
import { Entity } from '@hateoas-ts/resource';
import { Project } from './project.js';

export type KnowledgeGraphNode = {
  logicalEntityId: string;
  type: string;
  subType?: string;
  name: string;
  label?: string;
  description?: string;
};

export type KnowledgeGraphEdge = {
  diagramId: string;
  sourceLogicalEntityId: string;
  targetLogicalEntityId: string;
  relationType: string;
};

export type KnowledgeGraph = Entity<
  {
    projectId: string;
    nodes: KnowledgeGraphNode[];
    edges: KnowledgeGraphEdge[];
  },
  {
    self: KnowledgeGraph;
    project: Project;
  }
>;

File: libs/backend/domain/src/main/java/reengineering/ddd/teamai/model/KnowledgeGraph.java
package reengineering.ddd.teamai.model;

import java.util.List;

public record KnowledgeGraph(String projectId, List<Node> nodes, List<Edge> edges) {
  public record Node(
      String logicalEntityId,
      String type,
      String subType,
      String name,
      String label,
      String description) {}

  public record Edge(
      String diagramId,
      String sourceLogicalEntityId,
      String targetLogicalEntityId,
      String relationType) {}
}
